lib/geordi/cucumber.rb

```diff
@@ -32,33 +40,43 @@ def serial_execution_command
       unless argv.include?('--format') || argv.include?('-f')
         format_args = spinner_available? ? ['--format', 'CucumberSpinner::CuriousProgressBarFormatter'] : ['--format', 'progress']
       end
+      if argv.include?('rerun')
+        filter_command_line_features!
+      end
       [ Util.binstub_or_fallback('cucumber'), format_args, escape_shell_args(argv)].flatten.compact.join(' ')
     end
 
     def parallel_execution_command
       Interaction.note 'Using parallel_tests'
-      self.argv = argv - command_line_features
+      filter_command_line_features!
 
       type_arg = Util.gem_version('parallel_tests') > Gem::Version.new('0.7.0') ? 'cucumber' : 'features'
       features = features_to_run
       features = find_all_features_recursively('features') if features.empty?
-
       [
         'bundle exec parallel_test -t ' + type_arg,
-        %(-o '#{command_line_options.join(' ')}'),
+        %(-o "#{command_line_options.join(' ')}"),
         "-- #{features.join(' ')}",
       ].compact.join(' ')
     end
 
+    def filter_command_line_features!
+      self.argv = argv - command_line_features
+    end
+
     def escape_shell_args(*args)
       args.flatten.collect do |arg|
         arg.gsub(/([\\ "])/) { |_match| "\\#{Regexp.last_match(1)}" }
       end
     end
 
     def show_features_to_run
-      if command_line_options.include? 'rerun'
+      tags = command_line_options & ['--tags','-t']
+      if command_line_options.include?('rerun')
         Interaction.note 'Rerunning failed scenarios'
+      elsif command_line_options.include?('--tags') || command_line_options.include?('-t')
+        tag_index = command_line_options.find_index(tags.first) + 1
+        Interaction.note "Only features matching tag option #{command_line_options[tag_index]}"
```

Comment:
L74 should be moved above L78 IMHO. Alternatively, you could introduce a method `#command_line_tag_options` that even accounts for multiple tags options:
```
def command_line_tag_options
  [].tap do |tag_options|
    command_line_options.each_cons do |option, tags|
      tag_options << tags if option =~ /--tags|-t/
    end
  end
end
```

Reply:
I like your suggested method, good idea :+1: